packages/fxa-profile-server/lib/routes/profile.js

```diff
@@ -57,6 +60,16 @@ module.exports = {
         result.sub = creds.user;
       }
 
+      // Need to filter subscriptions by client ID for the request, since ALL
```

Comment:
This is the essence of the profile-server caching bug that I ran into:

We want to filter subscription capabilities by the client ID requesting the profile. We did this at the auth-server level. So, profile-server only got data for the current request's client ID.

But, profile-server caches this data keyed on UID, ignoring client ID. Thus, it would retain only the first request for a profile UID on a cache miss - regardless of client ID. So, if a different client later requested the same profile, it would get the cached data for the previous client - and thus not receive the correct subscription capabilities.

The change here is for profile-server receive and cache ALL subscription capabilities by client ID from auth-server. Then, at request time, filter the cached data for requesting client ID. 

This also keeps the profile cache invalidation code simple, because those parts of profile-server also only know about profile UID and not client IDs.